book/basics/units_basics.nim

```diff
@@ -0,0 +1,150 @@
+the navigators thought they read 1 N⋅s from the craft's sensors it was in fact 1 lbf⋅s.
+
+As an introduction to `Unchained`, let's find out what 1 lbf⋅s is in N⋅s to understand how much of a difference it made.
+"""
+nbCode:
+  import unchained
+
+  ## Define composite units
+  defUnit(N•s)
+  defUnit(lbf•s)
+
+  ## Assign variables
+  let lbfs = 1.lbf•s
+  let Ns = lbfs.to(N•s)
+
+  echo lbfs, " equals ", Ns
+
+nbText: """
+As you can see, they thought the value they read was 4.4 times smaller than it really was! 
+
+But what does the code do? Let's dissect it block-by-block:
+
+1. `defUnit(N•s)` - There are a ridiculous amount of combinations of base units.
+Therefore you must define composite units manually before you use them. More on this in a later section.
+2. `let lbfs = 1.lbf•s` - We use the dot (`.`) to assign a unit to a number. In this case we assign a variable with the value "1 lbf•s".
+3. `let Ns = lbfs.to(N•s)` - We use the `to` template to convert a variable from one unit to another.
+"""
+
+nbText: """
+## Defining and assigning units
+
+The first thing you probably will want to do is to associate a value with a unit, for example 1 Newton. There are multiple ways
+to specify a unit, both short and long versions:
+"""
+
+nbCodeInBlock:
+  ## Short version
+  let n1 = 1.N
+  ## Long version
+  let n2 = 1.Newton
+  ## Short version with prefix (milli)
+  let n3 = 1.mN
+  ## Long version with prefix
+  let n4 = 1.MilliNewton
+
+nbText: """
+That was easy enough, and being able to use prefixes makes for a saner user experience.
+When it comes to composite types it gets a bit more complicated though. For example if we want to use a unit `kg•m•s⁻¹` we
+have a few more things to consider. First and foremost, composite types must either be defined in a `defUnit` or be used in a
+dot expression (eg `10.kg•m•s⁻¹`) *before* it can be used in other parts of the code. Here comes a few valid and invalid cases:
+"""
+
+nbCodeInBlock:
+  ## Correct way of doing it!
+  defUnit(kg•m•s⁻¹)
+  proc unitProc(k: kg•m•s⁻¹) =
+    echo k
+
+nbText: """
+  ```nim
+  ## Incorrect way of doing it! Missing `defUnit`
+  proc unitProc(k: kg•m•s⁻¹) =
+    echo k
+  ```
+"""
+
+#[
+nbCodeInBlock:
+  ## Correct way of doing it!
+  let a = 5.kg•m•s⁻¹
+  proc unitProc(k: kg•m•s⁻¹) =
+    echo k
+
+nbText: """
+```nim
+## Incorrect way of doing it! Missing first use in `.`
+proc unitProc(k: kg•m•s⁻¹) =
+  echo k
+```
+"""
+
+]#
+
+nbText: """
+There is also the `UnitLess` type which represents a quantity without a unit like for example a count or a percentage.
+It is used like the other units with the addition that `UnitLess` numbers can be passed to procs accepting `float`: 
+"""
+
+nbCodeInBlock:
+  proc f(x: float): float = x*x + x + 1
+  let ul = 100.UnitLess
+  echo f(ul)
+  # This will fail beacuse `x` isn't UnitLess:
+  # let x = 100.kg
+  # echo f(x)
```

Comment:
```suggestion
  ## This will fail beacuse `x` isn't UnitLess:
  ## let x = 100.kg
  ## echo f(x)
```

otherwise it doesn't show up in the final result.

Reply:
Oopsie! I was writing it using my new code block parser 🙃